package.json

```diff
@@ -6,7 +6,11 @@
   "version": "3.1.0-SNAPSHOT",
   "author": "Nuxeo",
   "license": "Apache-2.0",
+  "engines": {
+    "node": ">=10.23.0"
+  },
   "scripts": {
+    "postinstall": "check-engine",
```

Comment:
Taking into account that in order to build the project, you need to do:
1. npm install
2. npm run bootstrap
This means that we stop at the end of 1 but prevent 2.

Given we won't have a full installation anw, could we do it in preinstall instead?

Reply:
Since this actually prevents the build from happening, and is thus a behavior change, I would not backport this to any of the maintenance branches. If by any chance there's a weird version we did not test where this build currently works, we'd break it. I think for maintenance we would be ok with just documenting it (both on README and on the package.json, with the "engines" property. WDYT?

Reply:
>Given we won't have a full installation anw, could we do it in preinstall instead?

The problem with `preinstall` is that you wouldn't have any module available to perform this check. We could either:
a) Come up with our own script that wouldn't rely on any 3rd party module (installed through `npm i`);
b) Force `npm i [moduleName]` and then perform the check using that module (all in the `preinstall` script);
Not sure if either of the options is particularly better considering how simple the current mechanism is already.

>Since this actually prevents the build from happening

Actually it doesn't. Using `check-engine` just ends up logging a few extra lines after the `npm i`, without changing anything else in the usual execution of the install. If we wanted to halt the install completely we could've just used the `engine-strict=true` flag on the `.npmrc` file, without having to use a 3rd party module.

Reply:
> Actually it doesn't.

I tried `npm install` with node v8 installed, and it results in an error, exiting with status code 1. This will most likely prevent any following npm step from running on CI, such as `npm run bootstrap` (and that is what happens when you run `npm run nuke` locally, for example. Can you confirm you experience this locally?

Reply:
You're right, I reproduce the issue. Apparently they have an [`--ignore` flag to avoid that](https://github.com/mohlsen/check-engine/tree/master#cli), but it was only introduced in `1.10.0`, and we're trying to run `1.8.1`. For the sake of simplicity, I'll just redefine the `preinstall` script as:
```
"postinstall": "check-engine || true",
```
In theory, it should do the trick just the same. WDYT?

Reply:
That should work. But, assuming this https://github.com/nuxeo/nuxeo-elements/pull/351#discussion_r554941456 takes effect, we can probably even use the --ignore flag, no?

Reply:
I would like to unlock it for sure, my concern is that by doing so we wouldn't be able to `npm i` on `maintenance-2.2.x`, that says it supports `node` `>=6.0.0`. Or should we keep `1.8.1` on `maintenance-2.2.x` and unlock everywhere else?

Reply:
If our only concern is 2.2.x, which is hitting EOL, then I'd rather lock it there.

Reply:
Sounds fair! I'll do that.